management-gateway/src/main/java/org/frankframework/management/bus/JsonResponseMessage.java

```diff
@@ -15,13 +15,43 @@
 */
 package org.frankframework.management.bus;
 
-import org.springframework.http.MediaType;
+import java.io.ByteArrayOutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
 
 import org.frankframework.util.JacksonUtils;
+import org.springframework.http.MediaType;
+
+import jakarta.json.Json;
+import jakarta.json.JsonStructure;
+import jakarta.json.JsonWriter;
+import jakarta.json.JsonWriterFactory;
+import jakarta.json.stream.JsonGenerator;
 
 public class JsonResponseMessage extends StringResponseMessage {
+	private static final JsonWriterFactory WRITER_FACTORY;
+
+	static {
+		Map<String, Boolean> config = new HashMap<>();
+		config.put(JsonGenerator.PRETTY_PRINTING, false);
+		WRITER_FACTORY = Json.createWriterFactory(config);
+	}
 
 	public JsonResponseMessage(Object payload) {
 		super(JacksonUtils.convertToJson(payload), MediaType.APPLICATION_JSON);
 	}
+
+	public JsonResponseMessage(JsonStructure payload) {
+		super(jsonStructureToString(payload), MediaType.APPLICATION_JSON);
+	}
+
+	private static String jsonStructureToString(JsonStructure payload) {
+		ByteArrayOutputStream boas = new ByteArrayOutputStream();
+		try (JsonWriter jsonWriter = WRITER_FACTORY.createWriter(boas)) {
+			jsonWriter.write(payload);
+		}
+
+		return new String(boas.toByteArray(), StandardCharsets.UTF_8);
```

Comment:
```suggestion
		Writer writer = new StringWriter(8192);
		try (JsonWriter jsonWriter = WRITER_FACTORY.createWriter(writer)) {
			jsonWriter.write(payload);
		}

		return writer.toString();
```
(niet direct toepassen want dan kloppen je imports natuurlijk niet 😉 )

En als je het niet op die manier wil doen, dan misschien een goed idee om ook bij het aanroepen van `createWriter(...)` de charset alvast op te geven, zodat je zeker weet dat het allemaal UTF-8 is.

Reply:
Ja ik vind het foei lelijke code en ben helemaal niet blij dat het zo op deze manier moet. Maar ik heb in alle haast en mijn enthusiasme hier verder niet naar gekeken. Goede tip iig!